### PR TITLE
Added more power data to the atm90e32 component

### PR DESCRIPTION
**Pull request by CircuitSetup**
## Description:
This PR adds reactive power, power factor, and chip temp, data which would be very useful to see for power quality in addition to what is already available.

## Checklist:
  - [x] The code change is tested and works locally.
  - [x] Tests have been added to verify that the new code works (under `tests/` folder).

If user exposed functionality or configuration variables are added/changed:
  - [x] Documentation added/updated in [esphome-docs](https://github.com/esphome/esphome-docs).

### CODE REVIEW

File: esphome/components/atm90e32/sensor.py [hunk 1]
```diff
@@ -40,6 +43,7 @@
     cv.Optional(CONF_PHASE_B): ATM90E32_PHASE_SCHEMA,
     cv.Optional(CONF_PHASE_C): ATM90E32_PHASE_SCHEMA,
     cv.Optional(CONF_FREQUENCY): sensor.sensor_schema(UNIT_HZ, ICON_FLASH, 1),
+    cv.Optional(CONF_CHIP_TEMP): sensor.sensor_schema(UNIT_CELSIUS, ICON_THERMOMETER, 1),
```
Comment (OttoWinter) [inline esphome/components/atm90e32/sensor.py]: There's no such constant `CONF_CHIP_TEMP` defined anywhere, same for `CONF_REACT_POW`. Please declare them above `ATM90E32_PHASE_SCHEMA`.

Also, these should be called `chip_temperature` and `reactive_power` (long form)

File: esphome/components/atm90e32/sensor.py [hunk 2]
```diff
@@ -28,8 +29,10 @@
 
 ATM90E32_PHASE_SCHEMA = cv.Schema({
     cv.Optional(CONF_VOLTAGE): sensor.sensor_schema(UNIT_VOLT, ICON_FLASH, 2),
-    cv.Optional(CONF_CURRENT): sensor.sensor_schema(UNIT_AMPERE, ICON_FLASH, 2),
-    cv.Optional(CONF_POWER): sensor.sensor_schema(UNIT_WATT, ICON_FLASH, 2),
+    cv.Optional(CONF_CURRENT): sensor.sensor_schema(UNIT_AMPERE, ICON_CURRENT-AC, 2),
```
Comment (OttoWinter) [inline esphome/components/atm90e32/sensor.py]:
```suggestion
    cv.Optional(CONF_CURRENT): sensor.sensor_schema(UNIT_AMPERE, ICON_CURRENT_AC, 2),
```

File: esphome/components/atm90e32/sensor.py (hunk 1 above)
Comment (CircuitSetup) [reply esphome/components/atm90e32/sensor.py]: This is fixed

**Review by OttoWinter — APPROVED** [review]

File: esphome/components/atm90e32/sensor.py [hunk 3]
```diff
@@ -2,14 +2,17 @@
 import esphome.config_validation as cv
 from esphome.components import sensor, spi
 from esphome.const import \
-    CONF_ID, CONF_VOLTAGE, CONF_CURRENT, CONF_POWER, CONF_FREQUENCY, \
-    ICON_FLASH, UNIT_HZ, UNIT_VOLT, UNIT_AMPERE, UNIT_WATT
+    CONF_ID, CONF_VOLTAGE, CONF_CURRENT, CONF_POWER, CONF_POWER_FACTOR, CONF_FREQUENCY, \
+    CONF_TEMPERATURE, ICON_FLASH, ICON_LIGHTBULB, ICON_CURRENT_AC, ICON_THERMOMETER, \
```
Comment (OttoWinter) [inline esphome/components/atm90e32/sensor.py]:
```suggestion
    ICON_FLASH, ICON_LIGHTBULB, ICON_CURRENT_AC, ICON_THERMOMETER, \
```

File: esphome/components/atm90e32/atm90e32.h [hunk 4]
```diff
@@ -19,10 +19,15 @@ class ATM90E32Component : public PollingComponent,
...
   void set_current_sensor(int phase, sensor::Sensor *obj) { this->phase_[phase].current_sensor_ = obj; }
   void set_power_sensor(int phase, sensor::Sensor *obj) { this->phase_[phase].power_sensor_ = obj; }
+  void set_reactive_power_sensor(int phase, sensor::Sensor *obj) { this->phase_[phase].reactive_power_sensor_ = obj; }
+  void set_power_factor_sensor(int phase, sensor::Sensor *obj) { this->phase_[phase].power_factor_sensor_ = obj; }
   void set_volt_gain(int phase, uint16_t gain) { this->phase_[phase].volt_gain_ = gain; }
   void set_ct_gain(int phase, uint16_t gain) { this->phase_[phase].ct_gain_ = gain; }
 
   void set_freq_sensor(sensor::Sensor *freq_sensor) { freq_sensor_ = freq_sensor; }
+  void set_chip_temperature_sensor(sensor::Sensor *chip_temperature_sensor) {
+    chip_temperature_sensor_ = chip_temperature_sensor;
```
Comment (OttoWinter) [inline esphome/components/atm90e32/atm90e32.h]:
```suggestion
    chip_temperature_sensor_ = chip_temperature_sensor;
```

(remove trailing space)

**Review by OttoWinter — APPROVED** [review]
To make travis happy :)

**Comment by CircuitSetup** [comment]
Thanks, I was having trouble with travis there.

I'll get the docs updated shortly.

**Comment by CircuitSetup** [comment]
updated documentation: https://github.com/esphome/esphome-docs/pull/380

**Pull request merged**